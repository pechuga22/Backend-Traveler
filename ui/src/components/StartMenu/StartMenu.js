import React from 'react'
import { Header, Rating, Icon, Button, Menu } from 'semantic-ui-react'
import { Link } from 'react-router-dom'

import './StartMenu.less'

export default function StartMenu({ tour }) {
  const { _id, location, rating, ratingCount } = tour
  return (
    <Menu widths={2} fixed='bottom' secondary className='start'>
      <Menu.Item>
        <Header>
          <Icon name='map marker alternate' color='grey' />
          <Header.Content>{location}</Header.Content>
          <Header.Subheader>
            {rating && (
              <Rating icon='favorite' defaultRating={rating} maxRating={5} />
            )}
            <span className='rate'>{rating}</span>
            <span className='counter'>({ratingCount})</span>
          </Header.Subheader>
        </Header>
      </Menu.Item>
      <Menu.Item>
        <Link to={`/tours/${_id}/tourview`}>
          <Button color='pink' circular>
            <h2>START</h2>
          </Button>
        </Link>
      </Menu.Item>
    </Menu>
  )
}
