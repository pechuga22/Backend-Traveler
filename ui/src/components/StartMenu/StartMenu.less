@import '../../variables.less';

div.ui.fixed.menu.start {
  height: 93px;
  border-radius: 50px !important;
  box-shadow: 0 -3px 20px 0 rgba(0, 0, 0, 0.16);
  backdrop-filter: blur(10px);
  opacity: 0.9;
  background: #fff;
  div.content {
    font-size: 18px;
    font-weight: normal;
    color: @secondaryColor;
    padding: 0;
  }

  .sub.header {
    padding-top: 0.3rem;
  }

  .rate {
    color: @blueGray;
    padding-left: 5px;
  }
  .counter {
    font-size: 10px;
    color: @greySecondary;
    padding-left: 5px;
  }
  h2 {
    margin: 5px 0;
  }
  .map.marker.alternate.icon {
    font-size: 14px;
  }
  .button {
    margin-left: 3rem;
  }
}
