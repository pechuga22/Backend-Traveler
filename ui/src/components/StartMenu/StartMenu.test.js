import React from 'react'
import { shallow, mount } from 'enzyme'
import { MemoryRouter, Link } from 'react-router-dom'
import toJson from 'enzyme-to-json'

import StartMenu from './StartMenu'

const tour = {
  _id: '4912397a-023-4bad-87a8-cdf36e4e6c0c',
  title:
    'El Metro de Medellín es un sistema de transporte masivo que sirve directamente no sólo a Medellin',
  location: 'Cochabamba',
  rating: 3.7,
}

describe('./components/StartMenu/StartMenu.js', () => {
  describe('.render()', () => {
    it('should render the StartMenu component', () => {
      const component = shallow(<StartMenu tour={tour} />)
      expect(toJson(component)).toMatchSnapshot()
    })

    it('should find START text StartMenu component', () => {
      const component = shallow(<StartMenu tour={tour} />)
      const wrapper = component.find('h2')
      expect(wrapper.text()).toBe('START')
    })

    it('should include link to tourview', () => {
      const component = mount(
        <MemoryRouter>
          <StartMenu tour={tour} />
        </MemoryRouter>
      )
      expect(component.find(Link).props().to).toBe(`/tours/${tour._id}/tourview`)
    })
  })
})
