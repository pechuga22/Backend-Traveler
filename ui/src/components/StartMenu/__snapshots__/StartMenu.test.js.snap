// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`./components/StartMenu/StartMenu.js .render() should render the StartMenu component 1`] = `
<Menu
  className="start"
  fixed="bottom"
  secondary={true}
  widths={2}
>
  <MenuItem>
    <Header>
      <Icon
        as="i"
        color="grey"
        name="map marker alternate"
      />
      <HeaderContent>
        Cochabamba
      </HeaderContent>
      <HeaderSubheader>
        <Rating
          clearable="auto"
          defaultRating={3.7}
          icon="favorite"
          maxRating={5}
        />
        <span
          className="rate"
        >
          3.7
        </span>
        <span
          className="counter"
        >
          (
          )
        </span>
      </HeaderSubheader>
    </Header>
  </MenuItem>
  <MenuItem>
    <Link
      to="/tours/4912397a-023-4bad-87a8-cdf36e4e6c0c/tourview"
    >
      <Button
        as="button"
        circular={true}
        color="pink"
      >
        <h2>
          START
        </h2>
      </Button>
    </Link>
  </MenuItem>
</Menu>
`;
